config.json

```diff
@@ -1407,17 +1432,17 @@
       {
         "groupId": "org.jetbrains.kotlin",
         "artifactId": "kotlin-stdlib",
-        "version": "1.5.31",
-        "nugetVersion": "1.5.31.3",
+        "version": "1.6.0",
```

Comment:
I think we need to hold off on updating Kotlin to `1.6.0`. Looking at the diffs makes me think they may have changed how they are encoding their "unsigned" types, and we may need a fix there or else all of our Kotlin support around them will be incorrect.

Example: these changed from `uint` to `int` for a type called `UIntArray`:

#### Type Changed: Kotlin.UIntArray

Removed methods:

```csharp
public static uint Get (int[] _this, int index);
public static void Set (int[] _this, int index, uint value);
```

Added methods:

```csharp
public static int Get (int[] arg0, int index);
public static void Set (int[] arg0, int index, int value);
```

Reply:
Seems like API was changed:

#### Type Changed: Kotlin.UByteArray

Removed methods:

```csharp
public static byte Get (byte[] _this, int index);
public static void Set (byte[] _this, int index, byte value);
public virtual int Size ();
```

Decompiled:

```java
public final class UByteArray implements Collection<UByte>, KMappedMarker
{
    // removed
    
    public static final byte get-w2LRezQ(final byte[] arg0, final int index) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        return UByte.constructor-impl(arg0[index]);
    }
    
    public static final void set-VurrAj0(final byte[] arg0, final int index, final byte value) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        arg0[index] = value;
    }
    
    public static int getSize-impl(final byte[] arg0) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        return arg0.length;
    }
    
    public int getSize() {
        return getSize-impl(this.storage);
    }

    // removed
}
```


#### Type Changed: Kotlin.UIntArray

Removed interfaces:

```csharp
Java.Lang.IIterable
Java.Util.ICollection
```

Removed methods:

```csharp
public static uint Get (int[] _this, int index);
public static void Set (int[] _this, int index, uint value);
public virtual int Size ();
```

Decompiled:

```java
public final class UIntArray implements Collection<UInt>, KMappedMarker
{
    // removed

    public static final int get-pVg5ArA(final int[] arg0, final int index) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        return UInt.constructor-impl(arg0[index]);
    }
    
    public static final void set-VXSXFK8(final int[] arg0, final int index, final int value) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        arg0[index] = value;
    }
    
    // removed
}
```


#### Type Changed: Kotlin.ULongArray

Removed methods:

```csharp
public static ulong Get (long[] _this, int index);
public static void Set (long[] _this, int index, ulong value);
public virtual int Size ();
```

Decompiled:

```java
public final class ULongArray implements Collection<ULong>, KMappedMarker
{
    // removed
    
    public static final long get-s-VKNKU(final long[] arg0, final int index) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        return ULong.constructor-impl(arg0[index]);
    }
    
    public static final void set-k8EXiF4(final long[] arg0, final int index, final long value) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        arg0[index] = value;
    }
    
    public static int getSize-impl(final long[] arg0) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        return arg0.length;
    }
    
    public int getSize() {
        return getSize-impl(this.storage);
    }

    // removed
}
```



#### Type Changed: Kotlin.UShortArray

Removed methods:

```csharp
public static ushort Get (short[] _this, int index);
public static void Set (short[] _this, int index, ushort value);
public virtual int Size ();
```


Decompiled:

```java
public final class UShortArray implements Collection<UShort>, KMappedMarker
{
    // removed
    
    public static final short get-Mh2AYeg(final short[] arg0, final int index) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        return UShort.constructor-impl(arg0[index]);
    }
    
    public static final void set-01HTLdE(final short[] arg0, final int index, final short value) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        arg0[index] = value;
    }
    
    public static int getSize-impl(final short[] arg0) {
        Intrinsics.checkNotNullParameter(arg0, "arg0");
        return arg0.length;
    }
    
    public int getSize() {
        return getSize-impl(this.storage);
    }

    // removed
}
```